src/components/angular2-modal/models/dialog-ref.ts

```diff
@@ -1,20 +1,39 @@
 import { ComponentRef } from '@angular/core';
-import { PromiseWrapper } from '@angular/core/src/facade/promise';
+import { PromiseWrapper, PromiseCompleter } from '@angular/core/src/facade/promise';
 
 /**
  * API to an open modal window.
  */
 export class DialogRef<T> {
-    contentRef: ComponentRef<any>;
+    /** 
+     * The reference to the component ref.
+     * This can be return undefined, since the componentRef is set only when the modal is shown.
+     * Use componentRefPromise to get a promise that will resolve when the componentRet is set.
+     * @return {ComponentRef<any>}
+     */
+    get contentRef(): ComponentRef<any> { return this._contentRef; }
+    set contentRef(value: ComponentRef<any>) {
+        this._contentRef = value;
+        this._conponentRefDeferred.resolve(value);
+    }
+
+    /**
+     * A promise that is resolved when the component ref is set.
+     * @return {Promise<ComponentRef<any>>}
+     */
+    get contentRefPromise() { return this._conponentRefDeferred.promise; }
```

Comment:
It also possible not to use contentRef at all.
See this example: http://plnkr.co/edit/9HYPPK?p=preview

In all plugins I have a chain of Backdrop component -> Content Component
In this example I don't have a component chain, only Backdrop component.
Instead of Content component I use a TemplateRef supplied... this means no ContentRef.